src/main/core/logical_processor.rs

```diff
@@ -0,0 +1,183 @@
+/*
+
+* The Shadow Simulator
+* Copyright (c) 2010-2011, Rob Jansen
+* See LICENSE for licensing information
+*/
+use crate::cshadow;
+use crate::utility::perf_timer::PerfTimer;
+use crossbeam::queue::SegQueue;
+use std::convert::TryFrom;
+use std::convert::TryInto;
+use std::sync::Mutex;
+use std::time::Duration;
+
+/// A set of `n` logical processors
+pub struct LogicalProcessors {
+    lps: Vec<LogicalProcessor>,
+}
+
+impl LogicalProcessors {
+    pub fn new(n: usize) -> Self {
+        let mut lps = Vec::new();
+        for _ in 0..n {
+            lps.push(LogicalProcessor {
+                cpu_id: unsafe { cshadow::affinity_getGoodWorkerAffinity() },
+                ready_workers: SegQueue::new(),
+                done_workers: SegQueue::new(),
+                idle_timer: Mutex::new(PerfTimer::new()),
+            });
```

Comment:
Performance timers were a compile-time option in the C code since they added a noticeable amount of overhead. Are these performance timers on a hot path? If so, should we make these enabled only when a Cargo "perf-timers" feature is configured?

Reply:
Yeah, it doesn't make much difference in the current version since it's still up to the (C) caller to stop and start the timers. We'll need to tackle it eventually though; done.